### ISSUE TITLE
Person Regression is not working

### ISSUE THREAD

**Issue by muck27** [jac99/FootAndBall#7]
Hey @jac99 I retrained your model with the issia and spd datasets and I am amazed at the results. I wanted to train the model with my custom datasets captured from  2 x 90 degree cameras. I had annotated 1200 images for person and ball for this dataset in darknet format. I created an adapter for the darknet pipeline and trained the network. 

The network trained for 20 epochs for a total loss of 0.79 and the player location loss was 0.0016. The ball and person classification looks good but the regression is not learning for some reason. I have tried changing the player location loss from smooth l1 to mse loss as well to no avail. 

Anything wrong I am doing?

![Screenshot from 2022-05-23 15-22-53](https://user-images.githubusercontent.com/48056260/169793943-315661f7-4124-4915-912f-0e5a4cdea926.png)

**Comment by jac99** [comment]
Hi,
I suspect, there's a problem with a training ground truth - maybe there's some error when you convert player bounding boxes to tensors. You can try the following:
1. Visualize the ground truth data that comes from the Dataloader. Maybe there's an error in the Dataset __getitem__ or in data augmentation code, and bounding boxes ground truth in incorrectly encoded. Take the bounding boxes data from the batch returned by Dataloder and visualize them, to ensure they are correct.
2. Try to train your model with very small training set (one image only). The network should overfit to this tiny training set - the loss should decrease almost to zero and the predictions should be almost perfect. if they are not - I would bet there's a problem with bounding boxes ground truth (or in the procedure to decode the bounding boxes regressed by the network).
3.

**Comment by muck27** [comment]
Ok I will try them....Also a question: In the paper you have mentioned that the player bounding boxes should have less variance...the camera setup we are using will result in higher variance amongst players in different parts of the pitch...will this architecture predict player boxes of high variance?

**Comment by jac99** [comment]
In our architecture, a player classifier and regressor use features from one hierarchy level of the backbone network. As an input they use a feature map that has (H/16, W/16) resolution (where H and W are the height and width of an input image). Such a design is not well suited to detect objects with very large variance in the bounding box size. Detectors such as YOLO or SSD use features from different hierarchy levels, bo be able to detect very large objects (e.g. converting most of the image) and much smaller objects.

But I don't know how well our architecture will handle variance in the player bounding box size. It will handle some amount of variance. Unless players occupy a large part of the screen, it should work, But you need to verify it experimentally.

**Comment by muck27** [comment]
![Screenshot from 2022-05-25 16-36-20](https://user-images.githubusercontent.com/48056260/170248589-45784bdd-247c-4cb8-98d7-6f79ed31f328.png)
The network is able to regress for the person boxes....If there is anybody out there with a similar problem, please disable the crop augmentation from the training augmentation script. I can put in some time later to understand why the crop augmentation was not working. Probably it is not resizing the bounding box after the crop aug was added or something on those lines....but for now the results are good and we can close the issue
